apps/emqx_connector/test/emqx_connector_mysql_SUITE.erl

```diff
@@ -0,0 +1,143 @@
+        {ok, #{poolname => emqx_connector_mysql}},
+        emqx_connector_mysql:on_start(<<"emqx_connector_mysql">>, mysql_config())
+    ),
+    Config.
+
+end_per_testcase(_, _Config) ->
+    ?assertEqual(
+        ok,
+        emqx_connector_mysql:on_stop(<<"emqx_connector_mysql">>, #{poolname => emqx_connector_mysql})
+    ).
+
+% %%------------------------------------------------------------------------------
+% %% Testcases
+% %%------------------------------------------------------------------------------
+
+% Simple test to make sure the proper reference to the module is returned.
+t_roots(_Config) ->
+    ExpectedRoots = [{config, #{type => {ref, emqx_connector_mysql, config}}}],
+    ActualRoots = emqx_connector_mysql:roots(),
+    ?assertEqual(ExpectedRoots, ActualRoots).
+
+% Not sure if this level of testing is appropriate for this function.
+% Checking the actual values/types of the returned term starts getting
+% into checking the emqx_connector_schema_lib.erl returns and the shape
+% of expected data elsewhere.
+t_fields(_Config) ->
+    Fields = emqx_connector_mysql:fields(config),
+    lists:foreach(
+        fun({FieldName, FieldValue}) ->
+            ?assert(is_atom(FieldName)),
+            if
+                is_map(FieldValue) ->
+                    ?assert(maps:is_key(type, FieldValue) and maps:is_key(default, FieldValue));
+                true ->
+                    ?assert(is_function(FieldValue))
+            end
+        end,
+        Fields
+    ).
+
+% Execute a minimal query to validate connection.
+t_basic_query(_Config) ->
+    ?assertMatch(
+        {ok, _, [[1]]},
+        emqx_connector_mysql:on_query(
+            <<"emqx_connector_mysql">>, {sql, test_query()}, undefined, #{
+                poolname => emqx_connector_mysql
+            }
+        )
+    ).
+
+% Perform health check.
+t_do_healthcheck(_Config) ->
+    ?assertEqual(
+        {ok, #{poolname => emqx_connector_mysql}},
+        emqx_connector_mysql:on_health_check(<<"emqx_connector_mysql">>, #{
+            poolname => emqx_connector_mysql
+        })
+    ).
+
+% Perform healthcheck on a connector that does not exist.
+t_healthceck_when_connector_does_not_exist(_Config) ->
+    ?assertEqual(
+        {error, health_check_failed, #{poolname => emqx_connector_mysql_does_not_exist}},
+        emqx_connector_mysql:on_health_check(<<"emqx_connector_mysql_does_not_exist">>, #{
+            poolname => emqx_connector_mysql_does_not_exist
+        })
+    ).
+
+% %%------------------------------------------------------------------------------
+% %% Helpers
+% %%------------------------------------------------------------------------------
+
+mysql_config() ->
+    #{
+        auto_reconnect => true,
+        database => <<"mqtt">>,
+        username => <<"root">>,
+        password => <<"public">>,
+        pool_size => 8,
+        server => {?MYSQL_HOST, ?MYSQL_PORT},
+        ssl => #{enable => false}
+    }.
+
+mysql_bad_config() ->
```

Comment:
Small nit: this is apparently unused.

Reply:
You're right. When I made a previous change this no longer is used. Removing.